### PR TITLE
adding filter on sobjects to mitigate issue with VF limit

### LINKED ISSUE #1223: BUG: Collection Size Exceeds 1,000 in new wizard
# Problem
when hitting the button to load the new wizard in a real client org we hit the Collection Size limit in VF

image

# Solution

# Logic Assertions
Steps to test the solution works:
- run the new wizard in an org with over 1,000 objects, or somehow mock the describe object list
- confirm no list size error is received

# QA  Proof it passed QA (steps and pictures)

# Implementation

### PR DESCRIPTION
**Pull request by seanpat09**
# Critical Changes
Mitigates bug in #1223 by filtering out EntityDefinitions where the following are false:
IsCustomizable
IsApexTriggerable
IsCustomSetting

# Changes

# Issues Closed
#1223

### CODE REVIEW

**Comment by Szandor72** [comment]
Can we please doublecheck the early return doesn't make the additional filtering obsolete? It looks to me that way

Can we also check the use of global describe - if we filter the results at one place, how about the others where the DescribeResult is used? 

I cannot estimate if the additional check would prevent specific (rare) standard objects from being used by DLRS. I trust you on that :)

**Comment by seanpat09** [comment]
The initial early return is to check if there are still too many sObjects even after doing the check in the `EntityDefinition` table. I was trying to change as a little as possible so I didn't want to remove the reference to `sobjectDescription`, which is caching `Schema.getGlobalDescribe()`. filtering in the for loops is done because we're still iterating over `sObjectDescription`. Also those for loops do some other stuff with the describe information that I wasn't sure was available from the EntityDefinition table.

As for the other uses of the DescribeResult, I believe we should be fine because we're not looping over all the results, but just using the a single value in the map retrieved by whatever the user had selected.

I am not sure if there are any standard objects that are blocked by these changes either, but it looks like like the regular standard objects are there.

**Comment by sfenton3** [comment]
Testing: Some objects have inordinately large number of child records. User object has the same number of child records as parent records. Something to keep note of when we make more changes to the new wizard.

Security feedback: 
I previously received feedback to put hardcoded SOQL into a selector. If not, we should have `With Security Enforced` because this is periodically reviewed for security concerns on the app exchange, and may raise a flag.

`List<EntityDefinition> entityDefinitions = [
          SELECT QualifiedApiName
          FROM EntityDefinition
          WHERE
            IsCustomSetting = FALSE
            AND IsCustomizable = TRUE
            AND isApexTriggerable = TRUE
        ];`

File: dlrs/main/classes/ManageLookupRollupSummariesNewController.cls [hunk 1]
```diff
@@ -21,7 +21,29 @@ public with sharing class ManageLookupRollupSummariesNewController {
...
+    get {
+      if(rollupableSObjectsNames == null) {
+        rollupableSObjectsNames = new Set<String>();
+
+        List<EntityDefinition> entityDefinitions = [
+          SELECT QualifiedApiName FROM EntityDefinition
+          WHERE IsCustomSetting = false
+          AND IsCustomizable = true
+          AND isApexTriggerable = true
+        ];
```
Comment (sfenton3) [inline dlrs/main/classes/ManageLookupRollupSummariesNewController.cls:36]: May want to add WITH SECURITY ENFORCED

**Review by sfenton3 — DISMISSED** [review]

**Review by sfenton3 — APPROVED** [review]

**Pull request merged**